### PR TITLE
8344299: SM cleanup in javax.naming modules

### PR DESCRIPTION
**Pull request by AlekseiEfimov**
The proposed change cleans-up `SecurityManager`, `doPriviledged`, and `AccessController` usages from `java.naming`, `jdk.naming.rmi` and `jdk.naming.dns` modules.

One noteworthy change: The `java.naming.rmi.security.manager` JNDI environment property can be used to setup the `SecurityManager`: if the property is added to an environment map the `RegistryContext` attempts to setup the `SecurityManager`. The environment property and related method are removed in this PR. 

`tier1` - `tier3`, and other non-tiered JNDI tests showed no failures related to this change.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344299](https://bugs.openjdk.org/browse/JDK-8344299): SM cleanup in javax.naming modules (**Bug** - P4)

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)
 * [Daniel Fuchs](https://openjdk.org/census#dfuchs) (@dfuch - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22421/head:pull/22421` \
`$ git checkout pull/22421`

Update a local copy of the PR: \
`$ git checkout pull/22421` \
`$ git pull https://git.openjdk.org/jdk.git pull/22421/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22421`

View PR using the GUI difftool: \
`$ git pr show -t 22421`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22421.diff">https://git.openjdk.org/jdk/pull/22421.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22421#issuecomment-2505038850)
</details>

### CODE REVIEW

**Comment by AlanBateman** [comment]
Good cleanup. Maybe for a follow on PR but I assume VersionHelper can be removed. There's been several post JEP 486 removals that mean that createThread and loadClass shouldn't be needed now. Also isSerialDataAllowed could move to Obj, which I think is the only place where we need to know if javaXXX attributes can be deserialised.

**Comment by AlekseiEfimov** [comment]
> Good cleanup. Maybe for a follow on PR but I assume VersionHelper can be removed. There's been several post JEP 486 removals that mean that createThread and loadClass shouldn't be needed now. Also isSerialDataAllowed could move to Obj, which I think is the only place where we need to know if javaXXX attributes can be deserialised.

I think we can do it as part of this PR: I have removed the LDAP `VersionHelper` (`com.sun.jndi.ldap.VersionHelper`) - it involved to put a replacement for `VersionHelper.loadClass` into a couple of LDAP-related classes. Pushed as 3c8687c17eb46ffae73c108f14377439e36ec518

`tier1` to `tier3` tests are still good with the latest change

File: src/java.naming/share/classes/com/sun/jndi/ldap/LdapDnsProviderService.java [hunk 1]
```diff
@@ -50,25 +47,10 @@ final class LdapDnsProviderService {
...
-            final PrivilegedAction<ServiceLoader<LdapDnsProvider>> pa =
-                    () -> ServiceLoader.load(
-                            LdapDnsProvider.class,
-                            ClassLoader.getSystemClassLoader());
-
-            providers = AccessController.doPrivileged(
-                pa,
-                null,
-                new RuntimePermission("ldapDnsProvider"),
-                SecurityConstants.GET_CLASSLOADER_PERMISSION);
```
Comment (AlanBateman) [inline src/java.naming/share/classes/com/sun/jndi/ldap/LdapDnsProviderService.java:70]: Can you check if SecurityConstants.GET_CLASSLOADER_PERMISSION be removed now, I think this is the last usage.

**Review by AlanBateman — APPROVED** [review]
I did a pass over the updated changes and don't see any issues, good cleanup.

**Comment by AlanBateman** [comment]
> I'm thinking we could simply remove SecurityConstants when nothing references it anymore - and not bother with removing its fields in the meantime. Would you advise removing its other possibly unused fields?

Sean has hollowed out SecurityConstants in pull/22418 but had to leave GET_CLASSLOADER_PERMISSION because of this usage. There will be many rounds of cleanup so doesn't matter if it comes later. ALL_PERMISSION will probably be the last one, don't know if Sean will get to this before the fork.

**Comment by AlekseiEfimov** [comment]
> Sean has hollowed out SecurityConstants in pull/22418 but had to leave GET_CLASSLOADER_PERMISSION because of this usage. There will be many rounds of cleanup so doesn't matter if it comes later. ALL_PERMISSION will probably be the last one, don't know if Sean will get to this before the fork.

Got it thanks. I'll remove `GET_CLASSLOADER_PERMISSION` since I'm removing its last usage. Combined with Sean changes in pull/22418 - SecurityConstants will only have `ALL_PERMISSION` and `PROVIDER_VER` left. Removed `GET_CLASSLOADER_PERMISSION` in d40273b22942fd3b59633ac3f67ceeac138c1768

**Review by AlanBateman — APPROVED** [review]
Great!

**Review by dfuch — APPROVED** [review]

**Pull request closed without merging**